### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by kentlouise**
## Issue Link
https://app.asana.com/0/1202107027507930/1202139395534037
https://app.asana.com/0/1202107027507930/1202139395534037

## Defintion of Done
 - Fetch Project Integration with tests
 - Google sign in and sign out tests

## Steps to reproduce
```
docker-compose exec frontend yarn test -- index.test.js
docker-compose exec backlog yarn test -- ProjectController.test.ts
```

## Test Cases
_backend_
 - [x] fetch all projects
 - [x] filter by provider
 - [x] search project

_frontend_
 - [x] fetch all projects
 - [x] filter by provider
 - [x] search project
 - [x] sign in
 - [x] sign out  

 _unit test be_
 - [x] fetch all projects
 - [x] filter by provider
 - [x] search project

- [x] refactored existing tests

## Notes
n/a

## Screenshots
image

### CODE REVIEW

**Comment by NielmarLaurente** [comment]
### Comment : 
- `Project_ID ` is not displayed in the List

image

File: fe/__tests__/login.test.js [hunk 1]
```diff
@@ -1,20 +1,71 @@
...
+    }
+    return <button onClick={handleClick}>{buttonText}</button>
+  }
 
-    const hasHeading = screen.getByRole('heading', {
-      name: /welcome to uni chart/i
+  return { GoogleLogin }
+})
+
+describe('when rendering login page', () => {
```
Comment (Alvilb) [inline fe/__tests__/login.test.js:32]: please change to capital W `When rendering login page` po

**Review by Alvilb — COMMENTED** [review]
there are failed tests
backlog
  - all passed
bff
  - 2 test suites failed
frontend
  - 2 test suites failed

for the failing test for detail page test I think its fine to ignore it since I fixed it in my pr. So just the snapshot test for frontend :bow:

**Comment by NielmarLaurente** [comment]
### Comment : 
- Reset should show the initial state of each filters like the first time you visit the page

https://user-images.githubusercontent.com/58454589/168963754-4fb60cff-4123-4184-847f-4ae4b768fc6b.mp4

File: fe/src/services/bffService.js [hunk 2]
```diff
@@ -35,3 +35,17 @@ export const getActiveSprintData = async (project_id, provider) => {
 
   return data
 }
+
+export const getProjects = async (params) => {
+  let data
+
+  await axios
+    .get(`${URL}projects`, {
+      params: params,
```
Comment (Alvilb) [inline fe/src/services/bffService.js:44]: here as well, please change to `params`

**Review by NielmarLaurente — APPROVED** [review]
## Approved PR #23 
### Working 👍 

- Can fetch all projects and providers
- Can be filter by provider
- Can be filter by project name
- Displayed Filtered data
- Can reset Filter
- Able to Logout
- Passed test for `frontend` and `backlog`

<details>

<summary> Screenshot </summary>

image

https://user-images.githubusercontent.com/58454589/168977252-6d4f7d41-3556-4204-9211-96135a6af2fc.mp4

image

image

</details>

**Testing Approval time : 0.5 hr
Total comments : 2
Total Iterations : 2**

**Comment by Alvilb** [comment]
please fix console errors. It happens if there are projects with same name
image

**Review by Alvilb — APPROVED** [review]
Approved!
- code is ok
- all tests passed
backlog:
image
bff
image
frontend:
note: test for detail page is failing (will be fixed in my pr)
image

**Pull request merged**